airbyte-integrations/connectors/source-domo/source_domo/source.py

```diff
@@ -0,0 +1,79 @@
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+
+import json
+from datetime import datetime
+from typing import Tuple, Optional, List, Any, Mapping
+
+from airbyte_cdk.logger import AirbyteLogger
+from airbyte_cdk.models import (
+    AirbyteCatalog,
+    AirbyteConnectionStatus,
+    AirbyteMessage,
+    AirbyteRecordMessage,
+    ConfiguredAirbyteCatalog,
+    Status,
+    Type,
+)
+from airbyte_cdk.sources import AbstractSource, streams
+
+from source_domo.client import DOMOClient
+from source_domo.streams import DOMOStream
+
+
+class SourceDomo(AbstractSource):
+    def check_connection(
+        self, logger: AirbyteLogger, config: Mapping[str, Any]
+    ) -> Tuple[bool, Optional[Any]]:
+        """
+        :param config: The user-provided configuration as specified by the source's spec. This usually contains information required to check connection e.g. tokens, secrets and keys etc.
+        :return: A tuple of (boolean, error). If boolean is true, then the connection check is successful and we can connect to the underlying data
+        source using the provided configuration.
+        Otherwise, the input config cannot be used to connect to the underlying data source, and the "error" object should describe what went wrong.
+        The error object will be cast to string to display the problem to the user.
+        """
+        try:
+            client_id = config["client_id"]
+            client_secret = config["client_secret"]
+            data_specs = config["data_specs"]
+
+            client = DOMOClient(client_id=client_id, client_secret=client_secret)
+            for name, data_spec in data_specs.items():
+                client.get_schema(data_spec["data_set"], data_spec["table_name"])
+                break
+            return (True, None)
+        except Exception as e:
+            return (False, e)
+
+    def streams(self, config: Mapping[str, Any]) -> List[DOMOStream]:
+        client_id = config["client_id"]
+        client_secret = config["client_secret"]
+        data_specs = config["data_specs"]
+
+        client = DOMOClient(client_id=client_id, client_secret=client_secret)
+        streams = []
+        for name, data_spec in data_specs.items():
+            streams.append(DOMOStream(client, data_spec))
+        return streams
```

Comment:
Is this dynamically creating/generating streams for each dataset in the incoming `data_specs` object? If so, we need to test that this is actually doing what it is supposed to in the Airbyte UI. Specifically, auth and schema reading/generation.

Reply:
It was. Refactored to hardcoded streams.